pkg/cache/ttl.go

```diff
@@ -63,7 +61,9 @@ func (c *ttlCache) put(key any, value any) {
 func (c *ttlCache) putWithTTL(key any, value any, ttl time.Duration) {
 	c.Lock()
 	defer c.Unlock()
-
+	if len(c.items) == 0 {
+		go c.doGC()
```

Comment:
There is a possibility that when `len(items) == 0`, the old goroutine has not exited, but a new put request arrives, which can lead to multiple `goGC()` goroutines existing for the same structure.

Reply:
there is one lock，the previous must exit when the item queue is empty

Reply:
It is possible during the gaps in gcInterval.

Reply:
I understand now. The deletion can only be done through goGC, and the situation I mentioned will not occur.

Reply:
you mean that the previous gc has deleted all items but not exit?

Reply:
1. remove all items(not through goGC)
2. put one item(a new goGC())
3. old goGC() tick

This situation can lead to unexpected goroutines.